### PR TITLE
[#159195774] Navigate back with Android back button

### PR DESCRIPTION
**Pull request by DenisFrezzato**
Testing the back navigation, I discovered that the `ROUTES.PIN_LOGIN` route is duplicated across `AppNavigator` and `PinNavigator`. Why `PinNavigator` should even exist?
I removed the `PinNavigator` and implemented this behaviour: ater the pin has been validated, the router resets the navigator stack, navigating to `ROUTES.MAIN`, so it's impossible to go back to pin screen.

### CODE REVIEW

File: ts/components/PreferenceItem.tsx [hunk 1]
```diff
@@ -4,8 +4,8 @@ import * as React from "react";
 import { connectStyle } from "native-base-shoutem-theme";
 import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
 
-import IconFont from "../components/ui/IconFont";
 import variables from "../theme/variables";
+import IconFont from "./ui/IconFont";
```
Comment (DenisFrezzato) [inline ts/components/PreferenceItem.tsx]: Somebody didn't run `prettier`. Should I revert this changes?

Comment (cloudify) [reply ts/components/PreferenceItem.tsx]: what do you mean?

Comment (DenisFrezzato) [reply ts/components/PreferenceItem.tsx]: Running `prettier` it changed several files I didn't edit. Should I revert the changes `prettier` applied or I leave them?

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>


New dependencies added: <a href='https://www.npmjs.com/package/recompose'><code>recompose</code></a> and <a href='https://www.npmjs.com/package/@types/recompose'><code>&#64;types/recompose</code></a>.


## Affected stories

  * 🐞 [#159195774](https://www.pivotaltracker.com/story/show/159195774): Come CIT quando tappo il back "hardware" da android voglio che il comportamento sia equivalente al back della navigation header



<h2><a href="https://github.com/acdlite/recompose">recompose</a></h2>
<p>Author: Andrew Clark</p>
<p>Description: A React utility belt for function components and higher-order components</p>
<p>Homepage: <a href="https://github.com/acdlite/recompose">https://github.com/acdlite/recompose</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 3 years ago</td></tr><tr><td>Last Updated</td><td>14 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>2</td></tr><tr><td>Releases</td><td>55</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/@babel/runtime'><code>&#64;babel/runtime</code></a>, <a href='https://www.npmjs.com/package/change-emitter'><code>change-emitter</code></a>, <a href='https://www.npmjs.com/package/fbjs'><code>fbjs</code></a>, <a href='https://www.npmjs.com/package/hoist-non-react-statics'><code>hoist-non-react-statics</code></a>, <a href='https://www.npmjs.com/package/react-lifecycles-compat'><code>react-lifecycles-compat</code></a> and <a href='https://www.npmjs.com/package/symbol-observable'><code>symbol-observable</code></a></td></tr><tr><td>Keywords</td><td>react, higher-order, components, microcomponentization, toolkit, utilities and composition</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

recompose
=========

[npm version](https://www.npmjs.com/package/recompose)

Recompose is a React utility belt for function components and higher-order components. See the [GitHub project page](https://github.com/acdlite/recompose) for more information.


</details>




<h2><a href="http://npmjs.com/package/@types/recompose">&#64;types/recompose</a></h2>
<p>Author: Unknown</p>
<p>Description: TypeScript definitions for Recompose</p>
<p>Homepage: <a href="http://npmjs.com/package/@types/recompose">http://npmjs.com/package/@types/recompose</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 2 years ago</td></tr><tr><td>Last Updated</td><td>14 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>23</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/@types/react'><code>&#64;types/react</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# Installation
> `npm install --save @types/recompose`

# Summary
This package contains type definitions for Recompose (https://github.com/acdlite/recompose).

# Details
Files were exported from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/recompose

Additional Details
 * Last updated: Mon, 06 Aug 2018 21:55:31 GMT
 * Dependencies: react
 * Global values: none

# Credits
These definitions were written by Iskander Sierra <https://github.com/iskandersierra>, Samuel DeSota <https://github.com/mrapogee>, Curtis Layne <https://github.com/clayne11>, Rasmus Eneman <https://github.com/Pajn>, Lucas Terra <https://github.com/lucasterra>.


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/PreferenceItem.tsx (hunk 1 above)
Comment (cloudify) [reply ts/components/PreferenceItem.tsx]: either you have a different prettier configuration or the tslint check in the circleci build failed to detect the prettier issues... anyway, you should not commit changes not related to the story you're fixing

**Comment by cloudify** [comment]
Thanks @DenisFrezzato , I'd like to keep this change on hold, it looks like we can't fix this in one single place but instead we have to change all screens?

**Comment by DenisFrezzato** [comment]
@cloudify Yes, it has to be applied to each screen of the navigator.

**Comment by francescopersico** [comment]
1) I can't test it now and i am reading the code with my smartphone but i have a doubt about the implementation provided in this PR.
What we have is a HOC that add the SAME IDENTICAL BEHAVIUOR to the wrapped component. I can't see any option in the HOC that can change that behaviour. Now my question is: why we can't just add that behaviour to the AppNavigator? In this way we don't need to wrap each screen.

2) The linked story is "Come CIT quando tappo il back "hardware" da android voglio che il comportamento sia equivalente al back della navigation header", now the back in the navigation header not always just do a "goBack" so the story is not really fulfilled
